packages/Operators/test/reference_implementation/DTK_ReferenceHexMesh.cpp

```diff
@@ -0,0 +1,231 @@
+//---------------------------------------------------------------------------//
+/*
+  Copyright (c) 2012, Stuart R. Slattery
+  All rights reserved.
+
+  Redistribution and use in source and binary forms, with or without
+  modification, are permitted provided that the following conditions are
+  met:
+
+  *: Redistributions of source code must retain the above copyright
+  notice, this list of conditions and the following disclaimer.
+
+  *: Redistributions in binary form must reproduce the above copyright
+  notice, this list of conditions and the following disclaimer in the
+  documentation and/or other materials provided with the distribution.
+
+  *: Neither the name of the University of Wisconsin - Madison nor the
+  names of its contributors may be used to endorse or promote products
+  derived from this software without specific prior written permission.
+
+  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+  A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+  HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+  SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+  LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+  DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+  THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+  (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+  OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*/
+//---------------------------------------------------------------------------//
+/*!
+ * \brief DTK_ReferenceHexMesh.cpp
+ * \author Stuart R. Slattery
+ * \brief Reference implementation of a parallel hex mesh for unit testing.
+ */
+//---------------------------------------------------------------------------//
+
+#include "DTK_ReferenceHexMesh.hpp"
+#include "DTK_ReferenceHex.hpp"
+#include "DTK_ReferenceNode.hpp"
+#include "DTK_ReferenceHexLocalMap.hpp"
+#include "DTK_ReferenceHexShapeFunction.hpp"
+#include "DTK_ReferenceHexIntegrationRule.hpp"
+#include <DTK_BasicEntitySet.hpp>
+#include <DTK_DBC.hpp>
+
+namespace DataTransferKit
+{
+namespace UnitTest
+{
+//---------------------------------------------------------------------------//
+// Hex mesh constructor.
+ReferenceHexMesh::ReferenceHexMesh(
+    const Teuchos::RCP<const Teuchos::Comm<int> >& comm,
+    const Teuchos::Array<double>& x_edges,
+    const Teuchos::Array<double>& y_edges,
+    const Teuchos::Array<double>& z_edges)
```

Comment:
The edges are the actual locations of the nodes in x, y, and z in a regular Cartesian grid. The size of these arrays equals the number of nodes in each direction.